### PR TITLE
feat: Adapt to `serverless@3` logging interface

### LINKED ISSUE #171: Serverless Framework v3: integrating with the new design
Hi 👋 , Serverless Framework core team here!

We're working on the v3 beta version right now. The v3 will include a redesigned CLI output:

image

You can try the v3 beta with: `npm i serverless@pre-3`

To help plugins integrate with that new design (and build a great user experience), we've added new APIs for plugins:

- [Documentation of the new API](https://www.serverless.com/framework/docs/guides/plugins/cli-output)
- [Video introduction (Youtube)](https://www.youtube.com/watch?v=K3IhZ9fImk4)

Any feedback is appreciated! If you want to update the plugin to use these new APIs, feel free to ping me if you have any questions!

If you don't have time to update the plugin, let me know, we can send a pull request!

### PR DESCRIPTION
**Pull request by pgrzesik**
## Description

Adapts to new logging API that is going to be introduced with `serverless@3`.

The change in scope of this plugin is fairly minor - I didn't want to change too much of the current logic, just to adapt it a bit to the new API. I'd be happy to hear about potential suggestions for improvements though 💯 

Closes: #171

## Checklist

* [x] CHANGELOG.md
* [x] README.md
* [x] Unit tests
* [x] Consider performance implications
* [x] Consider security implications

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -37,12 +37,21 @@ class ServerlessPlugin {
...
-  log(...args) {
+  log(msg) {
     if (this.config.logging) {
-      this.serverless.cli.log(...args);
+      if (this.v3Utils) {
+        if (msg.includes('WARNING')) {
+          this.v3Utils.log.warning(msg);
+        } else {
+          this.v3Utils.log.notice(msg);
+        }
```
Comment (neverendingqs) [inline src/index.js:51]: Doing a string search is probably fine for this PR, but I think it's best for the caller to decide between `.warning()` and `.notice()`.

Was there a similar `this.serverless.cli.warn()` in v2?

**Review by neverendingqs — APPROVED** [review]
LGTM thanks!

**Comment by neverendingqs** [comment]
Need to sign off, but will investigate build errors:

```
Run npm ci
added 61 packages in 0.696s
DOTENV: Loading environment variables from .env, .env.production:
	 - APP_MESSAGE
	 - REGION
	 - TEST_VAR
	 - STAGE
Environment: linux, node 12.22.8, framework 3.0.0-pre.3db0e613, plugin 6.0.0-pre.42186afc, SDK 4.3.0
Docs:        docs.serverless.com
Support:     forum.serverless.com

Bugs:        github.com/serverless/serverless/issues

Error:
Detected unrecognized CLI options: "--env".

1 deprecation found: run 'serverless doctor' for more details
Error: Process completed with exit code 1.
```

**Review by neverendingqs — CHANGES_REQUESTED** [review]
@pgrzesik - is `v3utils` available at this time?

The [docs](https://www.serverless.com/framework/docs/guides/plugins/cli-output) say they should exist, but when I tried it out on [neverendingqs:temp/v3utils](https://github.com/pgrzesik/serverless-dotenv-plugin/compare/adapt-to-v3-logging-interface...neverendingqs:temp/v3utils), it wasn't defined:

```
$ npx serverless@pre-3 package --stage production
v3Utils exists: false
Serverless: DOTENV: Loading environment variables from .env, .env.production:
Serverless: DOTENV (WARNING): if "include" is set, "exclude" is ignored.
```

I would like to understand why my test is not working as expected before merging if you have a chance.

**Comment by pgrzesik** [comment]
Hello @neverendingqs - thanks a lot for review, I was off for holidays but I'm back tomorrow and I'll see what might be going on in your case & in the build errors

File: src/index.js (hunk 1 above)
Comment (pgrzesik) [reply src/index.js:51]: No, there wasn't a `cli.warn` previously and only with this new logging mechanism we've introduced proper logging levels. 

I fully agree that called should be responsible for deciding the log level of the message, but on the other hand I wanted to keep the changes as small as possible. What do you think makes more sense from the perspective of the plugin here?

**Comment by pgrzesik** [comment]
Thanks for the review @neverendingqs - as for the build errors - they're caused by `--env` flag that is no longer supported in Framework `v3` - this is even listed in `README.md` that this flag will be no longer supported. 

As for your issue with `v3Utils` - I tried to reproduce but couldn't, for me it worked well, could you please check if you're not using some outdated version of `v3` prerelease?

**Review by neverendingqs — APPROVED** [review]
> Thanks for the review @neverendingqs - as for the build errors - they're caused by `--env` flag that is no longer supported in Framework `v3` - this is even listed in `README.md` that this flag will be no longer supported.

Thanks. Sorry I realized this between the two messages, and updated the README with this information.

> As for your issue with `v3Utils` - I tried to reproduce but couldn't, for me it worked well, could you please check if you're not using some outdated version of `v3` prerelease?

That was it thanks! `npx` was using a local copy of `serverless` which was at version 2.

File: src/index.js (hunk 1 above)
Comment (neverendingqs) [reply src/index.js:51]: I'm happy to do a refactor after merging this PR in. Thanks!

**Comment by pgrzesik** [comment]
Thanks a lot @neverendingqs 🙇

**Pull request merged**